lib/videojs-playlist-ui.js

```diff
@@ -1,25 +1,188 @@
 /*! videojs-playlist-ui - v0.0.0 - 2015-3-12
  * Copyright (c) 2015 Brightcove
  * Licensed under the Apache-2.0 license. */
-(function(window, videojs) {
-  'use strict';
-
-  var defaults = {
-        option: true
-      },
-      playlistUi;
-
-  /**
-   * Initialize the plugin.
-   * @param options (optional) {object} configuration for the plugin
-   */
-  playlistUi = function(options) {
-    var settings = videojs.util.mergeOptions(defaults, options),
-        player = this;
-
-    // TODO: write some amazing plugin code
-  };
-
-  // register the plugin
-  videojs.plugin('playlistUi', playlistUi);
-})(window, window.videojs);
+import * as playlist from 'videojs-playlist';
+
+const defaults = {
+  className: 'vjs-playlist'
+};
+
+const createThumbnail = function(thumbnail) {
+  if (!thumbnail) {
+    let placeholder = document.createElement('div');
+    placeholder.className = 'vjs-playlist-thumbnail';
+    return placeholder;
+  }
+
+  let picture = document.createElement('picture');
+  picture.className = 'vjs-playlist-thumbnail';
+
+  if (typeof thumbnail === 'string') {
+    // simple thumbnails
+    let img = document.createElement('img');
+    img.src = thumbnail;
+    picture.appendChild(img);
+  } else {
+    // responsive thumbnails
+
+    // additional variations of a <picture> are specified as
+    // <source> elements
+    for (let i = 0; i < thumbnail.length - 1; i++) {
+      let variant = thumbnail[i];
+      let source = document.createElement('source');
+      // transfer the properties of each variant onto a <source>
+      for (let prop in variant) {
+        source[prop] = variant[prop];
+      }
+      picture.appendChild(source);
+    }
+
+    // the default version of a <picture> is specified by an <img>
+    let variant = thumbnail[thumbnail.length - 1];
+    let img = document.createElement('img');
+    for (let prop in variant) {
+      img[prop] = variant[prop];
+    }
+    picture.appendChild(img);
+  }
+  return picture;
+};
+
+videojs.PlaylistMenuItem = videojs.Component.extend({
+  init: function(player, options) {
+    if (!options.item) {
+      throw new Error('Cannot construct a PlaylistMenuItem without an item option');
+    }
+    // stub out the element so Component doesn't construct one
+    options.el = true;
+    videojs.Component.call(this, player, options);
+    this.el_ = this.createEl(options.item);
+
+    this.item_ = options.item;
+    this.emitTapEvents();
+
+    this.on(['click', 'tap'], (event) => {
+      const currentIndex = player.playlist().indexOf(this.item_);
+      if (currentIndex >= 0) {
+        player.playlist.currentItem(currentIndex);
```

Comment:
you can probably pass in the currentIndex without any checks. We already do [range checks](https://github.com/brightcove/videojs-playlist/blob/master/src/playlist-maker.js#L55-L58).